runtime.md

```diff
@@ -66,13 +65,15 @@ This operation MUST generate an error if it is not provided the ID of a containe
 This operation MUST return the state of a container as specified in the [State](#state) section.
 In particular, the state MUST be serialized as JSON.
 
+After a user launches a container using the [start](#start) operation, state requests by the same user in the same namespaces with the same runtime installation MUST successfully return that container's state by the time the [pre-start hooks](config.md#prestart) are called (step (3) in the [lifecycle](#lifecycle)) through the final [post-stop hooks](#config#poststop) (step (8) in the [lifecycle](#lifecycle)).
+Runtimes MAY return state under other conditions, for example to a different user or when asked from a different namespace.
```

Comment:
On Mon, Feb 22, 2016 at 06:41:14AM -0800, Doug Davis wrote:
> > @@ -66,13 +65,15 @@ This operation MUST generate an error if it is not provided the ID of a containe
> >  This operation MUST return the state of a container as specified in the [State](#state) section.
> >  In particular, the state MUST be serialized as JSON.
> >  
> > +After a user launches a container using the [start](#start) operation, state requests by the same user in the same namespaces with the same runtime installation MUST successfully return that container's state by the time the [pre-start hooks](config.md#prestart) are called (step (3) in the [lifecycle](#lifecycle)) through the final [post-stop hooks](#config#poststop) (step (8) in the [lifecycle](#lifecycle)).
> > +Runtimes MAY return state under other conditions, for example to a different user or when asked from a different namespace.
> 
> This seems like a re-wording of the text above you removed.

It is.

> I'd prefer not to have text that's specific to gathering state since
> it applies to all ops, not just "getState()".

I'd rather explicitly make those other operations dependent on state
gathering [1].  You prefer not to make that dependency explicit [2],
but since I don't see a way around the dependency of some sort of
container ID → state registry, I'm not sure why you'd want to hide the
dependency in the spec.

[1]: https://github.com/wking/opencontainer-specs/commit/2fff5454fb7f20beec6110ff8725fc8507f4cce8#diff-b84a8d65d8ed53f4794cd2db7e8ea731R53
[2]: https://github.com/duglin/specs/pull/1/files#r53632688